src/Rules/Properties/ReadOnlyPropertyAssignRule.php

```diff
@@ -89,6 +91,10 @@ public function processNode(Node $node, Scope $scope): array
 				continue;
 			}
 
+			if ((new ObjectType(ArrayAccess::class))->isSuperTypeOf($propertyReflection->getNativeType())->yes()) {
+				continue;
```

Comment:
This is a wrong fix. You can't say that in general.

If I have `private readonly ArrayAcess $a` and I do `$this->a = new Foo()` twice, PHP does not allow that.

Reply:
🤦 of course - you're right.

I have pushed a patch to expand the test to test for your case and to also check that we're dealing with a `SetOffsetValueTypeExpr`